rs/tests/driver/src/k8s/tnet.rs

```diff
@@ -634,6 +651,185 @@ spec:
         }
         Ok(())
     }
+
+    pub async fn acquire_playnet_certificate(&self) -> Result<PlaynetCertificate> {
+        let client = Client::try_default().await?;
+        let config_map_api = Api::<ConfigMap>::namespaced(client.clone(), &TNET_NAMESPACE);
+
+        let playnet_prefix = format!("{}-playnet-", self.unique_name.clone().unwrap());
+        let config_map = (|| async {
+            let existing_playnets = config_map_api
+                .list(&Default::default())
+                .await?
+                .items
+                .into_iter()
+                .filter(|cm| {
+                    cm.metadata
+                        .labels
+                        .as_ref()
+                        .map(|l| l.contains_key(TNET_PLAYNET_LABEL))
+                        .unwrap_or_default()
+                })
+                .collect::<Vec<_>>();
+
+            if let Some(playnet) = existing_playnets.iter().find(|cm| {
+                cm.metadata
+                    .owner_references
+                    .as_ref()
+                    .expect("should have owner references")
+                    .iter()
+                    .any(|o| o.name == self.group_name)
+            }) {
+                slog::info!(
+                    self.logger.as_ref().unwrap(),
+                    "Using existing playnet: {}",
+                    playnet.name_any()
+                );
+                return Ok(playnet.clone());
+            }
+
+            if existing_playnets.len() >= PLAYNET_POOL_SIZE {
+                return Err(anyhow::anyhow!("Playnet pool is full"));
+            }
+
+            let random_number = (1..=PLAYNET_POOL_SIZE)
+                .into_iter()
+                .filter(|n| {
+                    !existing_playnets.iter().any(|cm| {
+                        cm.metadata
+                            .labels
+                            .clone()
+                            .expect("should have labels")
+                            .get(TNET_PLAYNET_LABEL)
+                            .unwrap()
+                            == n.to_string().as_str()
+                    })
+                })
+                .collect::<Vec<_>>()
+                .choose_multiple(&mut rand::thread_rng(), 1)
```

Comment:
Why choose a random number here instead of the first number like we do in Farm?

It's not that important but the advantage of choosing the first is that you usually get number 1 since not that many people use playnets. So it's a bit more predictable. In any case, I don't care that much about it and I'm just curious if there's a reason for using a random one.

Reply:
makes no difference to me. choosing random is slightly better to have a better chance avoiding race condition however unlikely. Do we care if the number is more predictable or not?

Reply:
Oh right race conditions. I now realise this chooses the playnet on the client's side instead of on the server's side where we used locking to prevent race conditions.